### PR TITLE
Serialize numbers properly in number validations

### PR DESCRIPTION
**Pull request by dmeijboom**
Introduce new 'Number' type to serialize integers properly for number validations.

Fixes: https://github.com/GREsau/schemars/issues/174

### CODE REVIEW

**Comment by dmeijboom** [comment]
Looks like the CI is failing on `tests/ui/repr_non_unit_variant.rs`. Specifically this part:

```
EXPECTED:
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
error: JsonSchema_repr: must be a unit variant
 --> tests/ui/repr_non_unit_variant.rs:7:5
  |
7 |     EmptyTuple(),
  |     ^^^^^^^^^^
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈

ACTUAL OUTPUT:
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
error: JsonSchema_repr: must be a unit variant
 --> tests/ui/repr_non_unit_variant.rs:7:5
  |
7 |     EmptyTuple(),
  |     ^^^^^^^^^^^^
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
```

That won't work with multiple Rust versions as the output differs. Not related to my changes though.

File: schemars/tests/ui/repr_non_unit_variant.stderr [hunk 1]
```diff
@@ -1,5 +1,5 @@
 error: JsonSchema_repr: must be a unit variant
- --> $DIR/repr_non_unit_variant.rs:7:5
+ --> tests/ui/repr_non_unit_variant.rs:7:5
```
Comment (jayvdb) [inline schemars/tests/ui/repr_non_unit_variant.stderr:2]: strange that other PRs are not failing here ; c.f. https://github.com/GREsau/schemars/actions

maybe this needs to be `$DIR` ?

**Comment by jayvdb** [comment]
I think there is a lot of overlap with https://github.com/GREsau/schemars/pull/182

**Comment by dmeijboom** [comment]
> I think there is a lot of overlap with #182

I agree. Seems like its a better implementation. Closing this in favor of #182.

**Pull request closed without merging**